lock-api-objects/src/test/java/com/palantir/lock/watch/LockEventTest.java

```diff
@@ -0,0 +1,231 @@
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.StringLockDescriptor;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.immutables.value.Value;
+import org.junit.Test;
+
+public class LockEventTest {
+    private static final String BASE = "src/test/resources/lock-event-wire-format/";
+    private static final boolean REWRITE_JSON_BLOBS = false;
+
+    private static final LockDescriptor LOCK_1 = StringLockDescriptor.of("abc");
+    private static final LockDescriptor LOCK_2 = StringLockDescriptor.of("test-lock");
+    private static final OldLockEvent OLD_LOCK_EVENT = ImmutableOldLockEvent.builder()
+            .sequence(10L)
+            .lockDescriptors(ImmutableSet.of(LOCK_1, LOCK_2))
+            .lockToken(LockToken.of(new UUID(1337, 42)))
+            .build();
+    private static final LockEvent BASELINE_LOCK_EVENT = ImmutableLockEvent.builder()
+            .sequence(10L)
+            .lockDescriptors(ImmutableSet.of(LOCK_1, LOCK_2))
+            .lockToken(LockToken.of(new UUID(1337, 42)))
+            .build();
+    private static final LockRequestMetadata LOCK_REQUEST_METADATA = LockRequestMetadata.of(ImmutableMap.of(
+            LOCK_1, ChangeMetadata.created(PtBytes.toBytes("new")), LOCK_2, ChangeMetadata.unchanged()));
+
+    // These are the mappers used by Conjure (minus the pretty-printing)
+    private static final ObjectMapper SERVER_MAPPER =
+            ObjectMappers.newServerObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    private static final ObjectMapper CLIENT_MAPPER =
+            ObjectMappers.newClientObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+
+    // This mapper is used to ensure that two JSONs are equal excluding indentation
+    private static final ObjectMapper VERIFYING_MAPPER = new ObjectMapper();
+    private static final Random RAND = new Random();
+
+    @Test
+    public void oldLockEventIsBaseline() {
+        assertSerializedEquals(OLD_LOCK_EVENT, "baseline");
+        assertDeserializedEquals("baseline", OLD_LOCK_EVENT, OldLockEvent.class);
+    }
+
+    @Test
+    public void serializedFormatIsUnchangedForAbsentMetadata() {
+        assertSerializedEquals(BASELINE_LOCK_EVENT, "baseline");
+        assertDeserializedEquals("baseline", BASELINE_LOCK_EVENT, LockEvent.class);
+
+        LockWatchEvent asLockWatchEvent = BASELINE_LOCK_EVENT;
+        assertSerializedEquals(asLockWatchEvent, "baseline");
+        assertDeserializedEquals("baseline", asLockWatchEvent, LockWatchEvent.class);
```

Comment:
I also serialize the event as `LockWatchEvent` to test if the json subtypes are handled correctly. Maybe this only needs a single dedicated test.